api-admin/src/main/java/bio/terra/pearl/api/admin/service/notifications/NotificationConfigExtService.java

```diff
@@ -70,19 +98,60 @@ public NotificationConfig replace(
     StudyEnvironment studyEnvironment =
         studyEnvironmentService.findByStudy(studyShortcode, environmentName).get();
     NotificationConfig existing = notificationConfigService.find(configId).get();
-
-    update.cleanForCopying();
-    update.setStudyEnvironmentId(studyEnvironment.getId());
-    update.setPortalEnvironmentId(portalEnvironment.getId());
-    if (update.getEmailTemplate() != null && update.getEmailTemplateId() == null) {
-      // this is a new email template, set the portal appropriately
-      update.getEmailTemplate().setPortalId(portalEnvironment.getPortalId());
-    }
-    NotificationConfig newConfig = notificationConfigService.create(update);
-
-    // once the new creation succeeds, deactivate the prior config.
+    authNotificationConfig(existing, portalEnvironment, studyEnvironment);
+    NotificationConfig newConfig = create(update, studyEnvironment, portalEnvironment);
+    // after creating the new config, deactivate the old config
     existing.setActive(false);
     notificationConfigService.update(existing);
     return newConfig;
   }
+
+  /**
+   * deactivates the notification config with configId, and adds a new config as specified in the
+   * update object. Note though that the portalEnvironmentId and studyEnvironmentId will be set from
+   * the portalShortcode and studyShortcode params. If the update contains a new email template,
+   * that template will be created as well.
+   */
+  @Transactional
+  public NotificationConfig create(
+      String portalShortcode,
+      String studyShortcode,
+      EnvironmentName environmentName,
+      NotificationConfig newConfig,
+      AdminUser user) {
+    authUtilService.authUserToPortal(user, portalShortcode);
+    PortalEnvironment portalEnvironment =
+        portalEnvironmentService.findOne(portalShortcode, environmentName).get();
+    authUtilService.authUserToStudy(user, portalShortcode, studyShortcode);
+    StudyEnvironment studyEnvironment =
+        studyEnvironmentService.findByStudy(studyShortcode, environmentName).get();
+
+    return create(newConfig, studyEnvironment, portalEnvironment);
+  }
+
+  /** confirms the given config is associated with the given study and portal environments */
+  private void authNotificationConfig(
```

Comment:
I called it auth since I was thinking of it in terms of this preventing unauthorized access of a config outside your study/portal.  But I think you're right that this method isn't limited to be used in auth contexts, so verify works